### DISCUSSION TITLE
Enhancement: Make CDN provider agnostic

### DISCUSSION THREAD

**Discussion by finnless** [danny-avila/LibreChat#1465] [Feature Requests & Suggestions]
### What features would you like to see added?

CDN support was implemented in #1438 but requires the usage of Firebase, which would lock the user into using Google services. This may not be possible for many user's use-cases. The core code should instead be abstracted to be provider-agnostic.

### More details

Could be done by defining a generic storage interface and implementing provider-specific classes.

### Which components are impacted by your request?

General, Endpoints, Plugins

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
It’s not intended to be the end-all be-all CDN provider. Similar to how there are many social oauth login providers, firebase is only one such provider of a CDN. The main difference is that you only need one. 

Most of the code can be copied and applied to another provider, and there is really only one missing piece for this to make this more modular, to determine which one to use, but I will configure it more clearly when another one is implemented. The initial implementation with firebase is to provide the groundwork of a CDN in place. 

With the update, I welcome anyone to try coding up another provider. At some point I will work on AWS myself
